jni/minicap-shared/aosp/src/minicap_31.cpp

```diff
@@ -0,0 +1,402 @@
+    mDesiredWidth = info.width;
+    mDesiredHeight = info.height;
+    mDesiredOrientation = info.orientation;
+    return 0;
+  }
+
+  virtual void
+  setFrameAvailableListener(Minicap::FrameAvailableListener* listener) {
+    mUserFrameAvailableListener = listener;
+  }
+
+  virtual int
+  setRealInfo(const Minicap::DisplayInfo& info) {
+    mRealWidth = info.width;
+    mRealHeight = info.height;
+    return 0;
+  }
+
+private:
+  int32_t mDisplayId;
+  uint32_t mRealWidth;
+  uint32_t mRealHeight;
+  uint32_t mDesiredWidth;
+  uint32_t mDesiredHeight;
+  uint8_t mDesiredOrientation;
+  android::sp<android::IGraphicBufferProducer> mBufferProducer;
+  android::sp<android::IGraphicBufferConsumer> mBufferConsumer;
+  android::sp<android::CpuConsumer> mConsumer;
+  android::sp<android::IBinder> mVirtualDisplay;
+  android::sp<FrameProxy> mFrameProxy;
+  Minicap::FrameAvailableListener* mUserFrameAvailableListener;
+  bool mHaveBuffer;
+  bool mHaveRunningDisplay;
+  android::CpuConsumer::LockedBuffer mBuffer;
+
+  int
+  createVirtualDisplay() {
+    uint32_t sourceWidth, sourceHeight;
+    uint32_t targetWidth, targetHeight;
+    android::status_t err;
+
+    switch (mDesiredOrientation) {
+    case Minicap::ORIENTATION_90:
+      sourceWidth = mRealHeight;
+      sourceHeight = mRealWidth;
+      targetWidth = mDesiredHeight;
+      targetHeight = mDesiredWidth;
+      break;
+    case Minicap::ORIENTATION_270:
+      sourceWidth = mRealHeight;
+      sourceHeight = mRealWidth;
+      targetWidth = mDesiredHeight;
+      targetHeight = mDesiredWidth;
+      break;
+    case Minicap::ORIENTATION_180:
+      sourceWidth = mRealWidth;
+      sourceHeight = mRealHeight;
+      targetWidth = mDesiredWidth;
+      targetHeight = mDesiredHeight;
+      break;
+    case Minicap::ORIENTATION_0:
+    default:
+      sourceWidth = mRealWidth;
+      sourceHeight = mRealHeight;
+      targetWidth = mDesiredWidth;
+      targetHeight = mDesiredHeight;
+      break;
+    }
+
+    // Set up virtual display size.
+    android::Rect layerStackRect(sourceWidth, sourceHeight);
+    android::Rect visibleRect(targetWidth, targetHeight);
+
+    // Create a Surface for the virtual display to write to.
+    MCINFO("Creating SurfaceComposerClient");
+    android::sp<android::SurfaceComposerClient> sc = new android::SurfaceComposerClient();
+
+    MCINFO("Performing SurfaceComposerClient init check");
+    if ((err = sc->initCheck()) != android::NO_ERROR) {
+      MCERROR("Unable to initialize SurfaceComposerClient");
+      return err;
+    }
+
+    // This is now REQUIRED in O Developer Preview 1 or there's a segfault
+    // when the sp goes out of scope.
+    sc = NULL;
+
+    // Create virtual display.
+    MCINFO("Creating virtual display");
+    mVirtualDisplay = android::SurfaceComposerClient::createDisplay(
+      /* const String8& displayName */  android::String8("minicap"),
+      /* bool secure */                 false
```

Comment:
this fails #34 and has to be set to true!

Reply:
ok, I see https://github.com/DeviceFarmer/minicap/issues/34#issuecomment-1403295652
+ https://github.com/Genymobile/scrcpy/issues/2129 << https://github.com/Genymobile/scrcpy/issues/36 

setting this to true will not help anything